java/vector/src/main/java/org/apache/arrow/vector/table/README.md

```diff
@@ -0,0 +1,384 @@
+// Ref count is still one, but there are two VSRs with a reference to myVector
+    vsr2.clear(); // Reference count for myVector is 0.
+```
+
+What is happening is that the reference counter works at a lower level than the VectorSchemaRoot interface. A reference counter counts references to ArrowBuf instances that control memory buffers. It doesn't count references to the ValueVectors that hold *them*. In the example above, each ArrowBuf is held by one ValueVector, so there is only one reference. This distinction is blurred though, when you call the VectorSchemaRoot's clear() method, which frees the memory held by each of the vectors it references even though another instance might refer to the same vectors.
+
+When you create Tables from vectors, it's assumed that there are no external references to those vectors. But, just to be on the safe side, the buffers underlying these vectors are transferred to new ValueVectors in the new Table, and the original vectors are cleared.
+
+*Don't do this either, but note the difference from above:*
+
+```Java
+IntVector myVector = createMyIntVector(); // Reference count for myVector = 1
+    Table t1 = new Table(myVector);  // myVector is cleared; Table t1 has a new hidden vector with
+    // the data from myVector
+    Table t2 = new Table(myVector);  // t2 has no rows because the myVector was just cleared
+    // t1 continues to have the data from the original vector
+    t2.clear();                      // no change because t2 is already empty and t1 is independent
+```
+
+With Tables, memory is explicitly transferred on instantiatlon so the buffers are held by that table are held by *only* that table.
+
+#### Creating Tables with dictionary-encoded vectors
+
+***Note: this section is highly speculative***
+
+Another point of difference is that dictionary-encoding is managed separately from VectorSchemaRoot, while Tables hold an optional DictionaryProvider instance. If any vectors in the source data are encoded, a DictionaryProvider must be set to un-encode the values.
+
+```java
+VectorSchemaRoot vsr = myVsr();
+    DictionaryProvider provider = myProvider();
+    Table t = new Table(vsr, provider);
+```
+
+In the immutable Table case, dictionaries are used in a way that's similar to the approach used with ValueVectors. To decode a vector, the user provides the dictionary id and the name of the vector to decode:
+
+```Java
+Table t = new Table(vsr, provider);
+    ValueVector decodedName = t.decode("name", 1L);
+```
+
+To encode a vector from a table, a similar approach is used:
+
+```Java
+Table t = new Table(vsr, provider);
+    ValueVector encodedName = t.encode("name", 1L);
+```
+
+***Current Limitation:*** One difference is the method that produces TSV formatted output has an extra switch instructing the Table to replace the encoded output with the decoded output where possible:
+
+```java
+String output = myTable.contentToTSVString(true);
+```
```

Comment:
It seems this doesn't exist in the code as written?

Reply:
True. I should remove that text until it gets implemented.

Reply:
removed.